src/api/ops-extrusions.js

```diff
@@ -58,6 +61,7 @@ const extrudeInPlane = function (cag, axis1, axis2, depth, options) {
  * (a 3D vector as a 3 component array)
  * @param {Boolean} [options.twiststeps=defaultResolution3D] - twiststeps determines the resolution of the twist (should be >= 1)
  * @param {Boolean} [options.twistangle=0] - twistangle The final face is rotated <twistangle> degrees. Rotation is done around the origin of the 2d shape (i.e. x=0, y=0)
+ * @param {CAG} [options.cag] - cag The target CAG to extrude to. Defaults to cag
```

Comment:
(I'm a github n00b and I think it ate my previous reply here so recreating; sorry if dup)

Renamed cag -> endShape.
This feature is for linear_extrude (which extrudes one CAG to another CAG). {rectangular,rotate}_extrude are about extruding a CAG along a Path, which is unrelated to this feature.
solidFromSlices would be much more computationally expensive and result in a more "pixelated" look than building up the walls. LMK if it's unclear why that is.

Reply:
Thanks for the changes, that was fast and looks good ! :-) I will double check on desktop, (code checks on phone are not ideal haha).
Would love an explanation about the "pixelation" you mention.
Btw solid from slices is quite buggy (weird polygon generation/ degenerate solids, lack of clarity about needing a polygon3d and not a cag as a base) and will git and overhaul for v2.

Reply:
What I mean by pixelation:

consider the 2-D case instead of 3-D for simplicity. We have points at (0,0) and (1,1) that we'd like to connect with a line. Drawing the diagonal directly between them is computationally simple and generates a smooth result. This is the analogue of what building walls is doing in 3D - directly describing the walls in a way that connects the relevant "sides" from the start & end CAGs.
On the other hand, consider building the same "diagonal" using a reduced-dimension solidFromSlices; that look like: choose a "resolution" parameter, and, starting at (0,0), alternately add horizontal and vertical segments (each of which is of length 1/resolution) to reach (1,1) via a "zigzag" pattern.  This is obviously visually less smooth than the diagonal approach above, and also is computationally more difficult. In the 3D case it's actually much worse because the shape of the intermediate slices are not simple, so requires interpolation at each step. (adding insult to injury for this unnecessary computation, this interpolation is precisely along the walls that could have been described directly :)).

Reply:
thanks a lot for the clarification @fischman ! you seem quite well versed in what is going on the the respective algorithms !
I can see a few caveats to trying to generalize your algorithm (just thinking aloud) to a set of interconnected 'segments' connecting different shapes.
hmm also your initial comment (in the PR) about this essentially creating an "almost" 3D convex hull (which is a good point given the result) also makes me wonder how we could regroup all of these into as least operations as possible
(one of the main aims of V2 is to reduce the clutter and clarify the api as much as possible). 
would you mind if we reach out to you for feedback on V2 while we develop?

Reply:
> I can see a few caveats to trying to generalize your algorithm (just thinking aloud) to a set of interconnected 'segments' connecting different shapes.

Sorry, I don't follow. Can you explain what the generalization you're thinking of is, and what issues you see with the current approach?

> hmm also your initial comment (in the PR) about this essentially creating an "almost" 3D convex hull (which is a good point given the result) also makes me wonder how we could regroup all of these into as least operations as possible

I don't think I used the word "almost" and I'm not sure what you're referring to.  I believe the result of this impl is in fact convex assuming its inputs are (each CAG is convex in 2d, and their sides being aligned enough to avoid self-intersection).
Maybe you're thinking that if one took the result of this extrude and union()'d it with some other solid, e.g. one that shared a face with the result of this extrusion, there may be opportunities to minimize the resulting object's complexity. 
I don't see off-hand a win here, but that doesn't mean much (I've only thought about it for a few minutes, and I don't know the innards of CSG.js well enough).

> (one of the main aims of V2 is to reduce the clutter and clarify the api as much as possible).
would you mind if we reach out to you for feedback on V2 while we develop?

Not at all, happy to help.